src/CoreWCF.CrankTests/CoreWCFPerfClient/Program.cs

```diff
@@ -0,0 +1,169 @@
+        public const string CloseAndOpen = "closeandopen";
+    }
+
+    public enum TestBinding { BasicHttp, NetTcp }
+
+    public class Program
+    {
+        private TestBinding _paramBinding = TestBinding.BasicHttp;
+        private TimeSpan _paramPerfMeasurementDuration = s_defaultPerfMeasurementDuration;
+        private string _paramServiceUrl = "";
+        private int _paramRequests = 0;
+        private bool _paramCloseAndOpen = false;
+
+        private readonly static TimeSpan s_defaultPerfMeasurementDuration = TimeSpan.FromSeconds(10);
+
+        public static async Task Main(string[] args)
+        {
+
+            Program test = new Program();
+
+            if (test.ProcessRunOptions(args))
+            {
+                var startTime = DateTime.Now;
+                int requestTime = 0;
+                int request = 0;
+
+                Binding binding = null;
+                switch (test._paramBinding)
+                {
+                    case TestBinding.BasicHttp:
+                        binding = new BasicHttpBinding();
+                        break;
+                    case TestBinding.NetTcp:
+                        binding = new NetTcpBinding(SecurityMode.None);
+                        break;
+                    default:
+                        break;
+                }
+
+                TimeSpan measurementDurationPerTime = test._paramPerfMeasurementDuration;
+
+                BenchmarksEventSource.Register("firstrequest", Operations.Max, Operations.Max, "First Request (ms)", "Time to first request in ms", "n0");
+                BenchmarksEventSource.Register("corewcfperf/requests", Operations.Max, Operations.Sum, "Requests (" + test._paramPerfMeasurementDuration.TotalMilliseconds + " ms)", "Total number of requests", "n0");
+                BenchmarksEventSource.Register("corewcfperf/rps/max", Operations.Max, Operations.Sum, "Requests/sec (max)", "Max requests per second", "n0");
+
+                startTime = DateTime.Now;
+                ChannelFactory<ISayHello> factory = new ChannelFactory<ISayHello>(binding, new EndpointAddress(test._paramServiceUrl));
+                factory.Open();
+                var client = factory.CreateChannel();
+                ((IClientChannel)client).Open();
+                var stopwatchFirstReq = new Stopwatch();
+                stopwatchFirstReq.Start();
+                var result = client.HelloAsync("hello world").Result;
+
+                BenchmarksEventSource.Measure("firstrequest", stopwatchFirstReq.ElapsedMilliseconds);
+
+                while (DateTime.Now <= startTime.Add(measurementDurationPerTime))
+                {
+                    var rtnResult = await client.HelloAsync("hello world");
+                    Console.WriteLine(rtnResult);
+                    request++;
+                    requestTime = request;
+                    if (requestTime >= test._paramRequests & test._paramCloseAndOpen)
+                    {
```

Comment:
Once request reaches test._paramRequests, it will create a new client every time as you are just copying request to requestTime. The resetting of requestTime to 0 at the end of the if block suggests you meant to simply increment requestTime instead of copying request across.